app/controllers/Application.scala

```diff
@@ -912,35 +914,34 @@ object Application extends Controller with Security {
     ).getOrElse(NotFound(views.html.static.trouble("No such subscriber plan: " + id)))
   }
 
-  val interestForm = Form(
-    single(
-      "scheme_id" -> number
-    )
-  )
-
-  def addSubscriberInterest(id: Int, action: String) = Action { implicit request =>
-    interestForm.bindFromRequest.fold(
-      errors => {
-        val sub = Subscriber.findById(id).get
-        BadRequest(views.html.subscriber.edit(sub, errors))
-      },
-      value => {
-        val sub2 = Subscriber.findById(id).get
-        val schemeId = interestForm.bindFromRequest.get
-        val scheme = Scheme.findById(schemeId).get
-        sub2.addInterest(scheme, action)
-        Redirect(routes.Application.editSubscriber(id))
-      }
-    )
-  }
-
-  def removeSubscriberInterest(id: Int, sid: Int) = Action { implicit request =>
+  def addSubscriberInterest(id: Int, schemeTag: String) = Action { implicit request =>
```

Comment:
Definitely - I'll do so. Didn't want to intrude on your work in this area as it was ongoing when this was added, but they should be protected routes